### DISCUSSION TITLE
Crazyswarm sudden Instability Mid Flight - using Optitrack

### DISCUSSION THREAD

**Discussion by 07hokage** [USC-ACTLab/crazyswarm#646] [Q&A]
I'm using Optitrack and VRPN ( hover_swarm.launch configured accordingly ). I was getting errors using optitrack closed source (Motive 1.10) hence switched to vrpn.  
Swam of 4 cfs were flying perfectly until 3 of them suddenly became unstable and started flying into the ground. 

Video here:  [https://youtu.be/WDlSzFrMwYM](https://youtu.be/WDlSzFrMwYM) 

I referred to the issue #529 . But in my case, it happened in the middle of the flight which is strange. I verified that the coordinate frames are correct and data is also matching the movement.  @whoenig can you please help me out. I'm really planning to procure more cfs if this can be sorted out

**Comment by 07hokage** [comment]
**Unpopular solution!** 
In addition to the above mentioned issue, I also experienced crashed while taking off and just after takeoff using multiple crazyflies. 
I observed that **LED blinking on the crazyflies was causing jittering and axis flipping**. When I covered the blinking Red LED, suddenly everything works fine now. I'm able to take off all 4 now and able to maneuver them.  
Can anyone confirm if they had tried this?

**Comment by 07hokage** [comment]
Update-2: It worked well for sometime, but started repeating gain. As soon as i send the takeoff command, the axes starts flipping and the  CF flips over

**Reply by whoenig** [reply]
I also always have this problem of sudden axis flips with OptiTrack. My only solution so far is to use the custom object tracking that the Crazyswarm provides, since this comes with a motion filter. Since you seem to have trouble getting this data (due to the old Motive version), you might need to implement your own motion filter, or only send position information to the drones (by setting the send_positions_only parameter in the launch file to True).

**Reply by 07hokage** [reply]
Thanks for the response @whoenig . I'll try to incorporate the mentioned changes and will experiment again.
